public/js/actions/sources.js

```diff
@@ -36,9 +40,44 @@ function queueSourcesDispatch(dispatch) {
  * Handler for the debugger client's unsolicited newSource notification.
  */
 function newSource(source) {
-  return ({ dispatch }) => {
-    if (isEnabled("features.sourceMaps") && source.sourceMapUrl) {
-      networkRequest(source.sourceMapUrl).then(res => {
+  return ({ dispatch, getState }) => {
+    if (isEnabled("features.sourceMaps") && source.sourceMapURL) {
+      const tab = getSelectedTab(getState());
+      const sourceMapURL = `${tab.get("url")}/${source.sourceMapURL}`;
+      networkRequest(sourceMapURL).then(res => {
+        Task.spawn(function* () {
+          yield dispatch(loadSourceText(source));
```

Comment:
This will give you the source text. Just do `const { text } = yield dispatch(loadSourceText(...))`